
From: Gerd Knorr <kraxel@bytesex.org>

Some minor changes for the ir-common module: Update for the RC5 keytable and
increase the IR_KEYTAB_SIZE #define.

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/drivers/media/common/ir-common.c |    4 ++--
 25-akpm/include/media/ir-common.h        |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff -puN drivers/media/common/ir-common.c~v4l-ir-common-update drivers/media/common/ir-common.c
--- 25/drivers/media/common/ir-common.c~v4l-ir-common-update	2004-06-19 13:58:13.914702520 -0700
+++ 25-akpm/drivers/media/common/ir-common.c	2004-06-19 13:58:13.918701912 -0700
@@ -75,6 +75,8 @@ IR_KEYTAB_TYPE ir_codes_rc5_tv[IR_KEYTAB
 	[ 0x35 ] = KEY_PLAY,            // play
 	[ 0x36 ] = KEY_STOP,            // stop
 	[ 0x37 ] = KEY_RECORD,          // recording
+	[ 0x3c ] = KEY_TEXT,            // teletext submode (Japan: 12)
+	[ 0x3d ] = KEY_SUSPEND,         // system standby
 
 #if 0 /* FIXME */
 	[ 0x0a ] = KEY_RESERVED,        // 1/2/3 digits (japan: 10)
@@ -106,8 +108,6 @@ IR_KEYTAB_TYPE ir_codes_rc5_tv[IR_KEYTAB
 	[ 0x39 ] = KEY_RESERVED,        // external 2
 	[ 0x3a ] = KEY_RESERVED,        // PIP display mode
 	[ 0x3b ] = KEY_RESERVED,        // view data mode / advance
-	[ 0x3c ] = KEY_RESERVED,        // teletext submode (Japan: 12)
-	[ 0x3d ] = KEY_RESERVED,        // system standby
 	[ 0x3e ] = KEY_RESERVED,        // crispener on/off
 	[ 0x3f ] = KEY_RESERVED,        // system select
 #endif
diff -puN include/media/ir-common.h~v4l-ir-common-update include/media/ir-common.h
--- 25/include/media/ir-common.h~v4l-ir-common-update	2004-06-19 13:58:13.915702368 -0700
+++ 25-akpm/include/media/ir-common.h	2004-06-19 13:58:13.919701760 -0700
@@ -27,7 +27,7 @@
 #define IR_TYPE_OTHER  99
 
 #define IR_KEYTAB_TYPE	u32
-#define IR_KEYTAB_SIZE	64  // enougth for rc5, probably need more some day ...
+#define IR_KEYTAB_SIZE	128  // enougth for rc5, probably need more some day ...
 
 #define IR_KEYCODE(tab,code)	(((unsigned)code < IR_KEYTAB_SIZE) \
 				 ? tab[code] : KEY_RESERVED)
_
